File: z/替换空格.cc
class Solution {
 public:
  void replaceSpace(char* str, int length) {
    if (str == NULL || length <= 0) {
      return;
    }
    int count = 0;
    int len = 0;
    while (str[len] != '\0') {
      if (str[len] == ' ') {
        ++count;
      }
      ++len;
    }
    int size = len + count * 2;
    if (size > length) {
      return;
    }
    str[size] = '\0';
    int i = size - 1;
    for (int j = len - 1; j >= 0; --j) {
      if (str[j] == ' ') {
        str[i] = '0';
        str[i - 1] = '2';
        str[i - 2] = '%';
        i -= 3;
      } else {
        str[i] = str[j];
        --i;
      }
    }
  }
};

File: h/替换空格.java
/*
请实现一个函数，将一个字符串中的每个空格替换成“%20”。
例如，当字符串为We Are Happy.则经过替换之后的字符串为We%20Are%20Happy
*/
public class Solution {
    public String replaceSpace(StringBuffer str) {
        String s = new String(str);
     
        s =  s.replaceAll(" ","%20");
        
        return s;
    }
}
